
syntax = "proto3";

package compr;


message FileMeta {
    string title = 1;
    string path = 2;
    string hash = 3;
}

message FileChunk {
    int32 offset = 1;
    bytes data = 2;
}

message SmallFileRequest {
    string uuid = 1;
    FileMeta meta = 2;
    FileChunk chunk = 3;
}

message FileRequest {
    string uuid = 1;
    oneof payload {
        FileMeta meta = 2;
        FileChunk chunk = 3;
    }
}

message FileResponse {
    int32 code = 1;
    string message = 2;
}

service Compressor {
    rpc CompressSmallFile(SmallFileRequest) returns (FileResponse) {}
    rpc CompressLargeFile(stream FileRequest) returns (FileResponse) {}
}